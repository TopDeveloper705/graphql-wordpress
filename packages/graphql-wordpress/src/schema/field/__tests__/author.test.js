import fields from 'field/author';
import getLoaders from 'data/loaders';

const args = {};
const context = {};
const info = {
  rootValue: {
    loaders: getLoaders(),
  },
};

describe('Test schema type field definition', () => {
  test('Test author field', () => {
    expect(fields.author.type.name).toMatchSnapshot();
  });

  test('Test resolve author field', async () => {
    const author = await fields.author.resolve({ author: 1 }, args, context, info);
    return expect(author).toMatchSnapshot();
  });

  test('Test resolve author field null', () => {
    expect(fields.author.resolve({ author: 0 }, args, context, info)).toBeNull();
  });
});
