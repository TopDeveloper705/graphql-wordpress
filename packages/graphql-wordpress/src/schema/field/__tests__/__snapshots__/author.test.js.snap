// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test schema type field definition Test author field 1`] = `"User"`;

exports[`Test schema type field definition Test resolve author field 1`] = `
User {
  "id": 1,
}
`;
